controller/api/destination/watcher/k8s.go

```diff
@@ -108,5 +111,15 @@ func InitializeIndexers(k8sAPI *k8s.API) error {
 		return fmt.Errorf("could not create an indexer for pods: %s", err)
 	}
 
+	err = k8sAPI.Srv().Informer().AddIndexers(cache.Indexers{ServerIndex: func(obj interface{}) ([]string, error) {
```

Comment:
I think this indexer is unused? It seems to only index by name/namespace anyway, which is unnecessary since the lister allows us to query this way directly.

Reply:
Ah yea it is. I was originally thinking we had to add this indexer in order to get the lister used in `endpoints_watcher.go`.